src/librustc_codegen_llvm/debuginfo/mod.rs

```diff
@@ -470,7 +470,9 @@ impl DebugInfoMethods<'tcx> for CodegenCx<'ll, 'tcx> {
                     match impl_self_ty.kind {
                         ty::Adt(def, ..) if !def.is_box() => {
                             // Again, only create type information if full debuginfo is enabled
-                            if cx.sess().opts.debuginfo == DebugInfo::Full {
+                            if cx.sess().opts.debuginfo == DebugInfo::Full
+                                && !impl_self_ty.needs_subst()
```

Comment:
Is this needed anymore? Also, should we preferring `has_param_types_or_consts` instead of `needs_subst`?

Reply:
I was able to remove this, fixed!